src/Language/Fixpoint/Solver/Solution.hs

```diff
@@ -309,8 +309,8 @@ addCEnv (x, be, bs) bs' = (x, be, F.unionIBindEnv bs bs')
 -- delCEnv :: F.IBindEnv -> CombinedEnv -> F.IBindEnv
 -- delCEnv bs (_, _, bs')  = F.diffIBindEnv bs bs'
 
-delCEnv :: Sol.Solution -> F.KVar -> F.IBindEnv -> CombinedEnv -> F.IBindEnv
-delCEnv s k bs (_, _, bs')  = F.diffIBindEnv bs (F.intersectionIBindEnv bs' _kbs)
+delCEnv :: Sol.Solution -> F.KVar -> F.IBindEnv -> F.IBindEnv
+delCEnv s k bs  = F.diffIBindEnv bs _kbs
```

Comment:
Since _kbs, the scope of k, is the intersection of all environments k appears in, the intersection with bs' seems redundant.